File: maxDepth.java
class Solution {
    public int maxDepth(String s) {
        int co =0;
        int maxi =0;
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
            if(ch=='('){
                co++;
                maxi = Math.max(maxi,co);
            }
            else if(ch==')'){
                co--;
            }
        }
        return maxi;
    }
}

File: maxDepth.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxi=0;
    int help(TreeNode* root,int co){
        if(root==nullptr) return 0;
        if(root->left==NULL && root->right==NULL) return co;
        int maxi = max(help(root->left,co+1),help(root->right,co+1));
        return maxi;
    }
    int maxDepth(TreeNode* root) {
        return help(root,1);
    }
};
